### PR TITLE
[Feature][Connector] JDBC source support partition

### LINKED ISSUE #1543: [Feature][Connector] JDBC source support partition
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

At now, flink jdbc support parallelism option, but not work. Need add partition column to support parallelism.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Hisoka-X**
## Purpose of this pull request
Add partition series option to support jdbc parallelism
This close #1543 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-jdbc/src/main/java/org/apache/seatunnel/flink/source/JdbcSource.java [hunk 1]
```diff
@@ -71,19 +80,17 @@
     private String password;
     private int fetchSize = DEFAULT_FETCH_SIZE;
     private Set<String> fields;
+    private int parallelism = 1;
```
Comment (ruanwenjun) [inline seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-jdbc/src/main/java/org/apache/seatunnel/flink/source/JdbcSource.java:83]: I find in the former code, the default value of parallelism in flink is -1. Why we change to 1?

Comment (ruanwenjun) [inline seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-jdbc/src/main/java/org/apache/seatunnel/flink/source/JdbcSource.java]: I find in the former code, the default value of parallelism in flink is -1. Why we change to 1?

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-jdbc/src/main/java/org/apache/seatunnel/flink/source/JdbcSource.java [hunk 2]
```diff
@@ -116,41 +128,86 @@ public void prepare(FlinkEnvironment env) {
...
             fetchSize = config.getInt(SOURCE_FETCH_SIZE);
         }
+        if (config.hasPath(PARALLELISM)) {
+            parallelism = config.getInt(PARALLELISM);
+        } else {
+            parallelism = env.getBatchEnvironment().getParallelism();
+        }
+        try {
+            Class.forName(driverName);
+            Connection connection = DriverManager.getConnection(dbUrl, username, password);
```
Comment (ruanwenjun) [inline seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-jdbc/src/main/java/org/apache/seatunnel/flink/source/JdbcSource.java]: The connection has not been closed here, it's better to close the connection in the finally or use try(Connection connection = xx)

**Review by ruanwenjun — COMMENTED** [review]
Just some little need to change, and we may also need to close the `JdbcInputFormat` in the close method.

**Comment by Hisoka-X** [comment]
> Just some little need to change, and we may also need to close the `JdbcInputFormat` in the close method.

Thanks for advise. Already fix it.

File: docs/en/flink/configuration/source-plugins/Jdbc.md [hunk 3]
```diff
@@ -8,16 +8,19 @@ Read data through jdbc
...
+|-----------------------| ------ | -------- | ------------- |
+| driver                | string | yes      | -             |
+| url                   | string | yes      | -             |
+| username              | string | yes      | -             |
+| password              | string | no       | -             |
+| query                 | string | yes      | -             |
+| fetch_size            | int    | no       | -             |
+| partition_column      | string | no       | -             |
+| partition_upper_bound | string | no       | -             |
+| partition_lower_bound | string | no       | -             |
```
Comment (ruanwenjun) [inline docs/en/flink/configuration/source-plugins/Jdbc.md]: Change type to long

File: docs/en/flink/configuration/source-plugins/Jdbc.md [hunk 4]
```diff
@@ -49,6 +52,18 @@ fetch size
...
 
+### partition_column [string]
+
+The column name for parallelism's partition, only support numeric type.
+
+### partition_upper_bound [int]
+
+The partition_column max value for scan, if not set SeaTunnel will query database get max value.
+
+### partition_lower_bound [int]
```
Comment (ruanwenjun) [inline docs/en/flink/configuration/source-plugins/Jdbc.md]: Change type to long.

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**